powerstore/resource_host.go

```diff
@@ -0,0 +1,273 @@
+package powerstore
+
+import (
+	"context"
+	"fmt"
+	"github.com/dell/gopowerstore"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"log"
+	"terraform-provider-powerstore/client"
+	"terraform-provider-powerstore/models"
+)
+
+// newHostResource returns host new resource instance
+func newHostResource() resource.Resource {
+	return &resourceHost{}
+}
+
+type resourceHost struct {
+	client *client.Client
+}
+
+// Metadata defines resource interface Metadata method
+func (r *resourceHost) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_host"
+}
+
+// Schema defines resource interface Schema method
+func (r *resourceHost) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+
+		MarkdownDescription: "Host resource",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "The unique identifier of the host.",
+				MarkdownDescription: "The unique identifier of the host.",
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				Description:         "Name of the host.",
+				MarkdownDescription: "Name of the host. This should be unique across all hosts in the cluster.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"description": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "Description of the host.",
+				MarkdownDescription: "Description of the host.",
+			},
+			"host_group_id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "Associated host group, if host is part of host group.",
+				MarkdownDescription: "Associated host group, if host is part of host group.",
+			},
+			"os_type": schema.StringAttribute{
+				Required:            true,
+				Description:         "Operating system of the host.",
+				MarkdownDescription: "Operating system of the host.",
+			},
+			"initiators": schema.SetNestedAttribute{
```

Comment:
Are we not considering CHAP-related params? https://developer.dell.com/apis/3898/versions/3.0.0.0/reference/openapi.json/paths/~1host~1%7Bid%7D/patch

Reply:
We are covering only the required parameters for a nested structure. CHAP-related parameters are optional but they can be added during update operation implementation.